internal/ent/hooks/notification.go

```diff
@@ -62,3 +64,53 @@ func isValidChannels(channels []enums.Channel) error {
 
 	return nil
 }
+
+// HookNotificationPublish runs after notification creation to publish to subscribers
+func HookNotificationPublish() ent.Hook {
+	return hook.On(func(next ent.Mutator) ent.Mutator {
+		return hook.NotificationFunc(func(ctx context.Context, m *generated.NotificationMutation) (generated.Value, error) {
+			// Execute the mutation first
+			val, err := next.Mutate(ctx, m)
+			if err != nil {
+				return nil, err
+			}
+
+			// After successful creation, publish to subscription manager
+			notification, ok := val.(*generated.Notification)
+			if !ok {
+				logx.FromContext(ctx).Warn().Msg("notification hook: value is not a notification")
+				return val, nil
+			}
+
+			logx.FromContext(ctx).Info().Str("notification_id", notification.ID).Msg("notification hook: notification created, attempting to publish")
```

Comment:
Let's switch this to debug; only needed in development and not production logs
```suggestion
			logx.FromContext(ctx).Debug().Str("notification_id", notification.ID).Msg("notification hook: notification created, attempting to publish")
```